### DISCUSSION TITLE
FATAL:  password authentication failed for user "teslamate"

### DISCUSSION THREAD

**Discussion by danshaw** [teslamate-org/teslamate#2857] [Getting started] [answered]
I'm installing this via docket on my Synology and accidentally the first time around placed a spaced in front of the password in my .yml file. After reading more about it, I stumbled upon [this answer](https://github.com/adriankumpf/teslamate/discussions/2773#discussioncomment-3191930), but I am having a hard time fixing it. Ideally I don't mind starting over again, but for some reason after deleting all of the containers, images, and folders, I still have an issue with postgres and it wanting to authenticate to the old password that I had messed up.

I guess my question is, how can I best get this working again? Any way to start over? Or is it best to change the password? If changing the password is the solution, can someone help me understand how to do that on a Synology? When I looked at all of the postgres users, I did not see a teslamate. So I am not sure how to reset it.

`[error] Postgrex.Protocol (#PID<0.165.0>) failed to connect: ** (Postgrex.Error) FATAL 28P01 (invalid_password) password authentication failed for user "teslamate"`

I should also mention that when I try to change the password of teslamate in postgres, I get:

`psql: FATAL:  Peer authentication failed for user "teslamate"`

I am running this using the root user.

**Comment by danshaw** [comment, marked as answer]
I was able to change the password by going into the postgres docker container. The challenge was that the Synology UI was not working, but when I did it via SSH, I was able to do so.

**Reply by instantchow** [reply]
@danshaw I'm also interested in running this docker-compose.yaml file on Synology. Would you be able to post a redacted version of your file? It might be helpful to other users. 

The one at https://docs.teslamate.org/docs/installation/docker/ seems out of date; The  [unofficial instructions](https://github.com/adriankumpf/teslamate/issues/113) point to editing (virtual_host) lines that don't exist in the docker-compose file in the official documentation site.

**Reply by danshaw** [reply]
The directions at https://docs.teslamate.org/docs/installation/docker/ are not actually out of date. At least they were not for me. Just make sure you create the local folders on your NAS to where those volumes should be.  As far as the network settings go, I didn't need to specify a virtual host variable. After running the compose, docker created its own bridged network and I can easily access Telsamate and Grafana using the IP and ports that are set.

<img width="1422" alt="image" src="https://user-images.githubusercontent.com/6180024/187229779-a221f7eb-d73b-4d75-a5b7-d38b73f9812a.png">

**Comment by tills13** [comment]
If anyone encounters a similar problem and they _swear_ they have all the right configuration, the password I was attempting to use had a `%` in it -- changing to a purely letters & numbers password worked for me.

**Reply by tills13** [reply]
It was possibly a quoting issue with my environment, actually. I had setup my systemd config like

```
...
Environment="DATABASE_PASS=asdasd%asdasd"
```
but I think I needed
```
...
Environment="DATABASE_PASS=asdasd%%asdasd"
```
